server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -118,6 +128,9 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion MERGE_ON_RECOVERY_VERSION = def(8_515_00_0, Version.LUCENE_9_11_1);
     public static final IndexVersion UPGRADE_TO_LUCENE_9_12 = def(8_516_00_0, Version.LUCENE_9_12_0);
     public static final IndexVersion ENABLE_IGNORE_ABOVE_LOGSDB = def(8_517_00_0, Version.LUCENE_9_12_0);
+
+    public static final IndexVersion UPGRADE_TO_LUCENE_10_0_0 = def(9_000_00_0, Version.LUCENE_10_0_0);
```

Comment:
@mark-vieira can you double check that this all looks correct to you. The parseUnchecked method is to parse older lucene versions until unsupported index versions are removed.

Reply:
@rjernst My understanding is that we didn't want to actually bump the major version on index or transport versions until we cut the branch for the final 8.x release. Is that correct?

Reply:
I was wondering about the version bump as well, thanks for calling it out explicitly.

Reply:
Yes, lucene 10 should have index version 9000000. Any index version changes that need to be backported to 8.x will need to be handled similar to transport version backports (ie very carefully)

Reply:
Transport Version - there are no serialization format changes in this PR, so no transport version change is necessary.

We absolutely need to an Index Version bump - because we're changing the index format.